### DISCUSSION TITLE
Deploying Remix's Developer blog example

### DISCUSSION THREAD

**Discussion by noghartt** [vercel/community#229] [Show & Tell]
Is there a way that I could do the deploying of [Remix's Developer blog](https://remix.run/docs/en/v1/tutorials/blog) example on Vercel without any problem? I wouldn't use a DB for it, I wanted to "fetch" the posts from the `posts/` folder using Node's `fs`. But I think that is impossible to do it? Or is there a way to handle it?

**Comment by leerob** [comment]
Hey there! You can [view a demo](https://remix-run-template.vercel.app/) of the Remix on Vercel starter or [deploy your own here](https://vercel.com/new/clone?repository-url=https://github.com/vercel/vercel/tree/main/examples/remix&template=remix).

> **Note:** [You _can_ include files in serverless functions](https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions), but I haven't tried it with this specific Remix example.

**Reply by noghartt** [reply]
> Note: You can include files in serverless functions, but I haven't tried it with this specific Remix example.

Yep! I tried to include files in serverless functions, but IDK if I'm doing this correctly.  I write something like this:

```json
// vercel.json
{
  "functions": {
    "api/index.js": {
      // ...
      "includeFiles": "posts/*.md"
    }
  }
}
```

But I think that something is going wrong or I'm doing something wrong because doesn't works :disappointed:
It continuing can't finding the `posts/*.md` file.

**Reply by leerob** [reply]
You could try using [mdx-bundler](https://github.com/kentcdodds/mdx-bundler) to fetch your markdown instead of reading from the FS. This still allows you to have the files in git, if you want.

**Reply by noghartt** [reply]
> You could try using [mdx-bundler](https://github.com/kentcdodds/mdx-bundler) to fetch your markdown instead of reading from the FS. This still allows you to have the files in git, if you want.

Yep! I turned to use MDX (Remix has implemented it out-of-the-box) and It's working correctly! Thank you so much!
But I  think that would be cool to use the default `posts/` folder as in the example, so I'll try some things to do it in the future

**Reply by techmind-git** [reply]
I tried the mdx-bundler approach on vercel myself but ran into issues with esbuild and the optional dependencies. That didn't quite work and I got this error https://github.com/evanw/esbuild/issues/1646. After downgrading esbuild that did not work with optional dependecies yet (to get around this for noew) I got another error described here: https://github.com/vitejs/vite/issues/1361. Locally all worked just fine. It took a little too much time so I abandoned this setup for now but both an example with local files and mdx-bundler might be benificial to more people.

**Reply by noghartt** [reply]
> I tried the mdx-bundler approach on vercel myself but ran into issues with esbuild and the optional dependencies. That didn't quite work and I got this error [evanw/esbuild#1646](https://github.com/evanw/esbuild/issues/1646). After downgrading esbuild that did not work with optional dependecies yet (to get around this for noew) I got another error described here: [vitejs/vite#1361](https://github.com/vitejs/vite/issues/1361). Locally all worked just fine. It took a little too much time so I abandoned this setup for now but both an example with local files and mdx-bundler might be benificial to more people.

I have the same issue with `esbuild`, I tested it on other platforms, and works perfectly. So, I think that is something wrong when deploying the Remix app on Vercel.

**Reply by jmn** [reply]
I had success with mdx-bundler and pinning `"esbuild": "0.12.09"` in my package.json. Later versions give me the error above (https://github.com/evanw/esbuild/issues/1646).

**Reply by brattonross** [reply]
Although it doesn't read from the filesystem, a solution that I think could work (inspired Kent C. Dodds' site) is fetching the blog posts from the repo using the GitHub API, and then compiling it with `mdx-bundler`.

To be able to work on files locally without having their content pushed to GitHub you could switch where the content is sourced from (fs or GitHub) based on an environment variable (e.g. `NODE_ENV`).

**Reply by fa7ad** [reply]
Did anyone ever find a way to get FS working? `vercel.json` errors out saying it can't find the function. While `mdx-bundler` does work, I'd prefer not using it
